syntax = "proto3";

package study_monitor.ipc;

import "google/protobuf/timestamp.proto";

message MonitorMsg {
  google.protobuf.Timestamp timestamp = 1;

  oneof payload {
    ConcentrationStatus concentration_status = 2;
    Error error = 3;
  }
}

message ConcentrationStatus  {
  float overall_score = 1;
  float sleeping_confidence = 2;
}

message Error {
  enum Type {
    UNKNOWN = 0;
  }

  Type type = 1;
  string msg = 2;
}
